src/dawn/IIR/IIR.proto

```diff
@@ -0,0 +1,314 @@
+//      Statement
+/* ===-----------------------------------------------------------------------------------------===*/
+
+// @brief Statement-Wrapper for the ASTStmt.
+//
+// This wrapper is used to have a common base for Stmts and Exprs
+message Statement {
+  // Singe ASTStmt in the Statement.
+  dawn.proto.statements.Stmt ASTStmt = 1;
+}
+
+
+/* ===-----------------------------------------------------------------------------------------===*/
+//      Accesses
+/* ===-----------------------------------------------------------------------------------------===*/
+// @brief Extent attached to a FieldAccess
+//
+// Each filed-access has extents in each dimension that define how far the accesses are in the
+//respective dimension
+message Extent {
+    int32 minus = 1;
+    int32 plus = 2;
+}
+
+message Extents {
+    // the three dimensions of an access-extent [Extent<i>, Extent<j>, Extent<k>]
+    repeated Extent extents = 1;
+}
+
+message Accesses {
+    // Map of all the write accesses [their accessID] to the Extents of the access
+    map<int32, Extents> writeAccess = 1;
+
+    // Map of all the read accesses [their accessID] to the Extents of the access
+    map<int32, Extents> readAccess = 2;
+}
+
+/* ===-----------------------------------------------------------------------------------------===*/
+//      Caches
+/* ===-----------------------------------------------------------------------------------------===*/
+message Window {
+    int32 minus = 1;
+    int32 plus = 2;
+}
+
+/// @brief Cache specification of cuda kernels (also used in Gridtools)
+message Cache {
+    // Available cache types
+    enum CacheType { CT_IJ = 0; CT_K = 1; CT_IJK = 2; CT_Bypass = 3; }
+
+    // IO policies of the cache
+    enum CachePolicy { CP_Unknown = 0; CP_FillFlush = 1; CP_Fill = 2; CP_Flush = 3; CP_EPFlush = 4;
+                       CP_BPFill = 5; CP_Local = 6; }
+    // Used Cache-Type
+    CacheType type = 1;
+
+    // Used IO-policy of the cace
+    CachePolicy policy = 2;
+
+    // Field-ID of the cached field
+    int32 accessID = 3;
+
+    // optional field: Interval on which cache is used
+    dawn.proto.statements.Interval interval = 4;
+
+    // optional field: Interval that is the union of all the accessIntervals
+    dawn.proto.statements.Interval enclosingAccessInterval = 5;
+
+    // optional field: Window on which we need to fill and flush
+    Window cacheWindow = 6;
+}
+
+/* ===-----------------------------------------------------------------------------------------===*/
+//      IIR-Tree
+/* ===-----------------------------------------------------------------------------------------===*/
+
+// @brief StatementAccessPair that is stored as the Leafs of the IIR
+message StatementAccessPair {
+    // the statement of the pair. To note that this can be a BlockStmt (or simiar) and therefore
+    // hold more than one stmt
+    Statement statement = 1;
+
+    // All the caller accesses of the statement
+    Accesses callerAccesses  = 2;
```

Comment:
Not sure I made a comment already about this. We said we wont support stencil functions in the IIR. So we dont need this separation of caller and callee here, isnt? 
Also the whole Accesses derived information? we can recompute while reconstructing the IIR

Reply:
what if stencils are calling other stencils?

Reply:
stencils calling other stencils is the first thing that is going to be inlined 
https://github.com/MeteoSwiss-APN/dawn/blob/master/src/dawn/Compiler/DawnCompiler.cpp#L143

And we should configure the passes in a way that if we request IIR, first the pass that inlines all the stencil functions is called

Reply:
done